src/highdicom/seg/sop.py

```diff
@@ -944,3 +966,1458 @@ def _encode_pixels(self, planes: np.ndarray) -> bytes:
+                            self.pixel_array
+                    else:
+                        pixel_array[out_frm, :, :, out_seg] = \
+                            self.pixel_array[seg_frame_ind, :, :]
+
+        if combine_segments:
+            # Check whether segmentation is binary, or fractional with only
+            # binary values
+            if self.segmentation_type != SegmentationTypeValues.BINARY:
+                is_binary = np.isin(
+                    np.unique(pixel_array),
+                    np.array([0.0, 1.0]),
+                    assume_unique=True
+                )
+                if not is_binary.all():
+                    raise ValueError(
+                        'Cannot combine segments if the segmentation is not'
+                        'binary'
+                    )
+
+            # Check for overlap by summing over the segments dimension
+            if np.any(pixel_array.sum(axis=-1) > 1):
+                raise RuntimeError(
+                    'Segments cannot be combined because they overlap'
+                )
+
+            # Scale the array by the segment number using broadcasting
+            if relabel:
+                pixel_value_map = np.arange(1, len(segment_numbers) + 1)
+            else:
+                pixel_value_map = segment_numbers
+            scaled_array = pixel_array * pixel_value_map.reshape(1, 1, 1, -1)
+
+            # Combine segments by taking maximum down final dimension
+            max_array = scaled_array.max(axis=-1)
+            pixel_array = max_array
+
+        return pixel_array
+
+    def get_source_sop_instances(self) -> List[Tuple[hd_UID, hd_UID, hd_UID]]:
+        """Get UIDs for all source SOP instances referenced in the dataset.
+
+        Returns
+        -------
+        List[Tuple[highdicom.UID, highdicom.UID, highdicom.UID]]
+            List of tuples containing Study Instance UID, Series Instance UID
+            and SOP Instance UID for every SOP Instance referenced in the
+            dataset.
+
+        """
+        return [self._ref_ins_lut[sop_uid] for sop_uid in self._ref_ins_lut]
+
+    def get_default_dimension_index_pointers(
+        self
+    ) -> List[BaseTag]:
+        """Get the default list of tags used to index frames.
+
+        The list of tags used to index dimensions depends upon how the
+        segmentation image was constructed, and is stored in the
+        DimensionIndexPointer attribute within the DimensionIndexSequence. The
+        list returned by this method matches the order of items in the
+        DimensionIndexSequence, but omits the ReferencedSegmentNumber
+        attribute, since this is handled differently to other tags when
+        indexing frames in highdicom.
+
+        Returns
+        -------
+        List[pydicom.tag.BaseTag]
+            List of tags used as the default dimension index pointers.
+
+        """
+        return self._dim_ind_pointers[:]
+
+    def dimension_indices_are_unique(
```

Comment:
I prefer function names to start with the verb

```suggestion
    def are_dimension_indices_unique(
```

Reply:
Resolved in 9f466a42080777e8943d4e00f5f2593a1977adaf